src/abstract-interpretation/data-frame/mappers/function-mapper.ts

```diff
@@ -141,13 +157,110 @@ function mapDataFrameConvert(
 }
 
 function mapDataFrameRead(
-	_args: readonly RFunctionArgument<ParentInformation>[],
-	_params: { fileName: FunctionParameterLocation, header: FunctionParameterLocation, separator: FunctionParameterLocation }
+	args: readonly RFunctionArgument<ParentInformation>[],
+	params: {
+		fileName:  FunctionParameterLocation,
+		header:    FunctionParameterLocation<boolean>,
+		separator: FunctionParameterLocation<string>,
+		quote:     FunctionParameterLocation<string>,
+		comment:   FunctionParameterLocation<string>,
+		skipLines: FunctionParameterLocation<number>,
+		text?:     FunctionParameterLocation
+	},
+	info: ResolveInfo
 ): DataFrameOperations[] {
+	const fileNameArg = getFunctionArgument(args, params.fileName, info);
+	const textArg = params.text ? getFunctionArgument(args, params.text, info) : undefined;
+	let source: string | undefined;
+	let request: RParseRequest | undefined;
+
+	if(fileNameArg !== undefined && fileNameArg !== EmptyArgument) {
+		const fileName = resolveIdToArgValue(fileNameArg, info);
+
+		if(typeof fileName === 'string') {
+			source = fileName;
+			const referenceChain = fileNameArg.info.file ? [requestFromInput(`file://${fileNameArg.info.file}`)] : [];
+			const sources = findSource(fileName, { referenceChain: referenceChain });
+
+			if(sources?.length === 1) {
+				source = sources[0];
+				request = getSourceProvider().createRequest(source);  // create request from resolved source file path
+			} else if(params.text === undefined && unescapeSpecialChars(fileName).includes('\n')) {
+				request = requestFromInput(unescapeSpecialChars(fileName));  // create request from string if file name argument contains newline
+			}
+		}
+	} else if(textArg !== undefined && textArg !== EmptyArgument) {
+		const text = resolveIdToArgValue(textArg, info);
+
+		if(typeof text === 'string') {
+			source = text;
+			request = requestFromInput(unescapeSpecialChars(text));
+		}
+	}
+	if(request === undefined) {
+		return [{
+			operation: 'read',
+			operand:   undefined,
+			args:      { source: source, colnames: undefined, rows: undefined }
+		}];
+	}
```

Comment:
Ist there a reason for why you do not outsource this into a obtainRequestFromRead?